### PR TITLE
fix wrong unread counter not being fixed

### LINKED ISSUE #4417: Counter values don't get recalculated anymore after the whole list is loaded.
1. Have wrong folder counter
2. Load all mails of that counter
3. Error -> Counter is not updated

Expect: Folder counter should be synchronized.

Hint:
* it may be possible to override counter values with the admin client to get to the broken state
* check if the count gets recalculated at all at the intended time
* try breaking something in the client to get to an invalid count

### Test Notes

you'll have to build a client that can set invalid mail counts (maybe expose the method that corrects it in MailListView.ts globally and build the client against the test system, call the method for the correct listId with an incorrect count and then check that the issue is fixed in the client under test.

* [ ] mail lists with a multiple of 100 mails and an incorrect mail counter are corrected when scrolled to the end
* [ ] mail list not containing a multiple of 100 mails and and incorrect mail counter are corrected when scrolled to the end
* [ ] mail lists with correct counters remain correct when scrolled to the end
* [ ] previous points work with the inbox
* [ ] previous points work with the trash 
* [ ] previous points work with a custom folder
* [ ] none of these lead to a neverending cycle of trying to fix and then re-dispatching (check console for "fixing up" or "no fixup needed" messages that signal the end of the cycle)

### PR DESCRIPTION
**Pull request by ganthern**
close #4417

### CODE REVIEW

**Comment by ganthern** [comment]
so, how do we test this anyway?

**Comment by charlag** [comment]
Write it incorrectly manually I guess

File: src/settings/EditNotificationEmailDialog.ts [hunk 1]
```diff
@@ -211,9 +212,7 @@ export function show(existingTemplate: NotificationMailTemplate | null, customer
 				customerProperties.getAsync().then(customerProperties => {
 					templates = customerProperties.notificationMailTemplates
 
-					if (
-						customerProperties.notificationMailTemplates.filter(t => t !== existingTemplate && t.language === selectedLanguageStream()).length > 0
-					) {
+					if (count(customerProperties.notificationMailTemplates, t => t !== existingTemplate && t.language === selectedLanguageStream()) > 0) {
```
Comment (charlag) [inline src/settings/EditNotificationEmailDialog.ts:215]: `.some()`

File: src/settings/emaildomain/VerifyOwnershipPage.ts [hunk 2]
```diff
@@ -89,7 +90,7 @@ export class VerifyOwnershipPageAttrs implements WizardPageAttrs<AddDomainData>
 					let customDomainInfos = getCustomMailDomains(this.data.customerInfo)
 
 					//domain is already assigned to this account
-					if (customDomainInfos.filter(domainInfo => domainInfo.domain === this.data.domain()).length) {
+					if (count(customDomainInfos, domainInfo => domainInfo.domain === this.data.domain())) {
```
Comment (charlag) [inline src/settings/emaildomain/VerifyOwnershipPage.ts:93]: `.some()`

File: src/gui/base/List.ts [hunk 3]
```diff
@@ -92,59 +92,58 @@ export interface ListConfig<T, R extends VirtualRow<T>> {
...
+	scrollUpdateLater: boolean = false
 	/**
 	 * The id of the timeout to updateDomElements if updateLater == true.
 	 */
 	private repositionTimeout: TimeoutID | null
 	/** sorted with _config.sortCompare */
-	readonly loadedEntities: T[] = []
+	private readonly loadedEntities: ElementType[] = []
+	/** the last time the set of entities in the loadedEntities list changed */
+	private lastElementChangedTime = window.performance.now()
```
[the remark this answers is not in the document]
Comment (charlag) [reply src/gui/base/List.ts:143]: We could just make some `{}`, reset it on update and compare identity.

**Review by charlag — APPROVED** [review]

**Pull request merged**